src/main/java/com/ericsson/ei/subscriptionhandler/InformSubscription.java

```diff
@@ -94,12 +102,13 @@ public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson)
         String subscriptionName = getSubscriptionField("subscriptionName", subscriptionJson);
         String notificationType = getSubscriptionField("notificationType", subscriptionJson);
         String notificationMeta = getSubscriptionField("notificationMeta", subscriptionJson);
-
         String key = "";
         String val = "";
         MultiValueMap<String, String> mapNotificationMessage = new LinkedMultiValueMap<>();
         ArrayNode arrNode = (ArrayNode) subscriptionJson.get("notificationMessageKeyValues");
-        if (arrNode.isArray()) {
+
+        boolean notificationMessageKeyValuesHasValues = !arrNode.toString().equals("[]");
+        if (notificationMessageKeyValuesHasValues) {
```

Comment:
if  you use this notificationMessageKeyValuesHasValues only one time, use !arrNode.toString().equals("[]") directly in if statement without creating variable.